### ISSUE TITLE
Pretrained Model

### ISSUE THREAD

**Issue by CuongNguyen218** [WongKinYiu/PartialResidualNetworks#9]
Hi @WongKinYiu , 
Did you provide pretrained model of darknet53-PRN?

**Comment by CuongNguyen218** [comment]
@WongKinYiu , and I have a question about your paper:
![image](https://user-images.githubusercontent.com/43350150/70388020-01fd4b80-19df-11ea-988d-5ef7f78dafba.png)
Can you explain this picture for me?
G1 Timestamp t = 1 start at Layer 5 right?
And i don't know why G2 at Layer 2 in t =2. I don't see any connection between L4 and L2.

**Comment by WongKinYiu** [comment]
shortcut/concatenate also will receive the gradient at the same time.
so G1 start at layers 1,3,5 (above connections are shortcuts).
then G2 will be transmitted to layers 2,4 (5➔4, 3➔2, 1➔).

**Comment by CuongNguyen218** [comment]
@WongKinYiu , Can you share darknet-53-prn and darknet53-fprn pretrained model on Imagenet for me ?

**Comment by WongKinYiu** [comment]
@CuongNguyen218 Hello, 
the model files are too big to upload to github.
and i do not have enough cloud drive space.

**Comment by CuongNguyen218** [comment]
@WongKinYiu ,
I have a unlimited google drive account. Please give me email address and i will send you my account. I need pretrained model for my bachelor thesis. Thank!

**Comment by WongKinYiu** [comment]
https://github.com/WongKinYiu/PartialResidualNetworks/blob/master/pdf/iccvw-prn.pdf

the first one is my e-mail address.

**Comment by CuongNguyen218** [comment]
I just send an email to x102432003@yahoo.com.tw Please check your mailbox.

**Comment by WongKinYiu** [comment]
i just upload the pre-trained weights.

**Comment by CuongNguyen218** [comment]
I got your pretrain on ImageNet. How can I used it for Yolov3 (convert to darknet53-conv74 format)

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by WongKinYiu** [comment]
https://github.com/AlexeyAB/darknet#how-to-train-tiny-yolo-to-detect-your-custom-objects

using partial.

**Comment by CuongNguyen218** [comment]
@WongKinYiu ,
I'm training yolov3 with your darknet53-prn weights and get some -nan on log. is it normal ?
![image](https://user-images.githubusercontent.com/43350150/70496876-d9429680-1b43-11ea-9e41-85bc9113ce5a.png)

**Comment by WongKinYiu** [comment]
@CuongNguyen218 

It seems there are no small objects in your own dataset.

**Comment by CuongNguyen218** [comment]
@WongKinYiu ,
![image](https://user-images.githubusercontent.com/43350150/70789336-65afbc00-1dc5-11ea-8922-473b830a48b2.png)
Can you explain the image above for me? Somethings wrong with the number of channel in the middle feature map. For example, 96 = 128 - (1 - 1/2)/2 *128 is not (1- gamma) / 2 * c in the picture. and why 1/8 appear in the red line below ?
![image](https://user-images.githubusercontent.com/43350150/70789910-89273680-1dc6-11ea-8117-f8e949df1d84.png)

**Comment by WongKinYiu** [comment]
![image](https://user-images.githubusercontent.com/12152972/70790769-b24cc480-1dd0-11ea-85e8-c09cd4ca2364.png)

there r 1/2/8/8/4 layers in 1/2/3/4/5 stages.
and...
![image](https://user-images.githubusercontent.com/12152972/70790989-18d1e280-1dd1-11ea-93fa-3dcbd122e533.png)

so in 3rd stage, c is 256 and γc is 128.
the channel number of 8 layers should be 240, 224, 208, 192, 176, 160, 144, 128.

**Comment by CuongNguyen218** [comment]
> ![image](https://user-images.githubusercontent.com/12152972/70790769-b24cc480-1dd0-11ea-85e8-c09cd4ca2364.png)
> 
> there r 1/2/8/8/4 layers in 1/2/3/4/5 stages.
> and...
> ![image](https://user-images.githubusercontent.com/12152972/70790989-18d1e280-1dd1-11ea-93fa-3dcbd122e533.png)
> 
> so in 3rd stage, c is 256 and γc is 128.
> the channel number of 8 layers should be 240, 224, 208, 192, 176, 160, 144, 128.

I know it when read your paper but I think your figure of characteristic 2 lead to misleading.

**Comment by CuongNguyen218** [comment]
@WongKinYiu , 
I try replace darknet53 with darknet 53-prn but my inference time is increase about 20% from 45 FPS to 55 FPS. How did you do to reduce inference time from 22ms to 11ms

**Comment by WongKinYiu** [comment]
55 FPS is faster than 45 FPS. The inference time is decrease, not increase...

and did u check the blops of ur implementation is same as our reported bflops (25.1 BFLOPs)?

**Comment by buiduchanh** [comment]
@CuongNguyen218 Hello, Can you share this link of darknet53-PRN pretrained weights ?

**Comment by zht022** [comment]
@CuongNguyen218  Hi, can you share your pretrained weights of darknet53-PRN with your google drive link?
@WongKinYiu  Or can you send the pretrained weights of darknet53-PRN to my email? My email address is zht022@mail.ustc.edu.cn 
Thanks!

**Comment by zht022** [comment]
@CuongNguyen218 Hi, can you share your pretrained weights of darknet53-PRN with your google drive link?
@WongKinYiu Or can you send the pretrained weights of darknet53-PRN to my email? My email address is zht1994@mail.ustc.edu.cn
Thanks!
